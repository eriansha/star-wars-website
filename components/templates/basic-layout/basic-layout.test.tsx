import { cleanup, render, screen } from '@testing-library/react'

import BasicLayout from './basic-layout'

const mockHeader = jest.fn()
const mockFooter = jest.fn()

jest.mock('@/organisms/header', () => () => {
  mockHeader()
  return <div>header</div>
})

jest.mock('@/organisms/footer', () => () => {
  mockFooter()
  return <div>footer</div>
})

const TestedComponent = () => (
  <BasicLayout>
    <h1>Rendering Layout</h1>
  </BasicLayout>
)

afterEach(() => {
  cleanup
  jest.clearAllMocks()
})

describe('Person Section', () => {
  it('renders without crashing', () => {
    render(<TestedComponent />)
  })

  it('render header and footer', () => {
    render(<TestedComponent />)
    expect(mockHeader).toHaveBeenCalledTimes(1)
    expect(mockFooter).toHaveBeenCalledTimes(1)
  })

  it('render children props', () => {
    const { getByRole } = render(<TestedComponent />)

    const childrentElem = screen.getByRole('heading', {
      name: 'Rendering Layout'
    })

    expect(getByRole('main')).toContainElement(childrentElem)
  })

  it('matches snapshot', () => {
    const tree = render(<TestedComponent />)
    expect(tree.container).toMatchSnapshot()
  })
})
