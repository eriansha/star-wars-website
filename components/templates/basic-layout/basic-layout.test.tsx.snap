// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Section matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen relative"
  >
    <div>
      header
    </div>
    <div
      class="flex-1"
    >
      <main>
        <h1>
          Rendering Layout
        </h1>
      </main>
      <div>
        footer
      </div>
    </div>
  </div>
</div>
`;
